config/v1/types_cluster_version_test.go

```diff
@@ -10,14 +10,44 @@ import (
 func TestKnownClusterVersionCapabilities(t *testing.T) {
 	exists := struct{}{}
 	known := make(map[ClusterVersionCapability]struct{}, len(KnownClusterVersionCapabilities))
-	for _, cap := range KnownClusterVersionCapabilities {
-		known[cap] = exists
+	for _, capability := range KnownClusterVersionCapabilities {
+		known[capability] = exists
 	}
 
 	for set, caps := range ClusterVersionCapabilitySets {
-		for _, cap := range caps {
-			if _, ok := known[cap]; !ok {
-				t.Errorf("Capability set %s contains %s, which needs to be added to KnownClusterVersionCapabilities", set, cap)
+		for _, capability := range caps {
+			if _, ok := known[capability]; !ok {
+				t.Errorf("Capability set %s contains %s, which needs to be added to KnownClusterVersionCapabilities", set, capability)
+			}
+		}
+	}
+}
+
+// TestSubsetsVersionCapabilities verifies that all preceding sets
+// (order given by capabilitySetOrder) are subsets of following sets
+func TestSubsetsVersionCapabilities(t *testing.T) {
+	// defines the order by which subsets are tested
+	var capabilitySetOrder = []ClusterVersionCapabilitySet{
+		ClusterVersionCapabilitySetNone,
+		ClusterVersionCapabilitySet4_11,
+		ClusterVersionCapabilitySet4_12,
+		ClusterVersionCapabilitySetCurrent,
```

Comment:
i don't think there's any guarantee that the sets only grow in size from version to version.

We could decide that some cap is no longer enabled by default, at some point (perhaps the cap gets deprecated/removed entirely).